Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/scan/VaccinationQrCodeScanViewModel.kt

```diff
@@ -21,14 +22,19 @@ class VaccinationQrCodeScanViewModel @AssistedInject constructor(
 
     val errorEvent = SingleLiveEvent<Throwable>()
 
+    val errorEventWithFaq = SingleLiveEvent<Throwable>()
+
     fun onScanResult(barcodeResult: BarcodeResult) = launch {
         try {
             event.postValue(Event.QrCodeScanInProgress)
             val qrCode = vaccinationQRCodeValidator.validate(barcodeResult.text)
             val vaccinationCertificate = vaccinationRepository.registerVaccination(qrCode)
             event.postValue(Event.QrCodeScanSucceeded(vaccinationCertificate.personIdentifier.codeSHA256))
         } catch (e: Throwable) {
-            errorEvent.postValue(e)
+            if (e is InvalidHealthCertificateException && e.showFaqButton) {
```

Comment:
Should we just check that before creating the dialog, and then only use the `errorEvent`, without the additional `errorEventWithFaq`?